node/rpc/src/wallet_api.rs

```diff
@@ -0,0 +1,255 @@
+pub(crate) async fn wallet_default_address<DB, KS>(
+    data: Data<State<DB, KS>>,
+) -> Result<String, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let keystore = data.keystore.read().await;
+
+    let addr = wallet::get_default(&*keystore)?;
+    Ok(addr.to_string())
+}
+
+/// Export KeyInfo from the Wallet given its address
+pub(crate) async fn wallet_export<DB, KS>(
+    data: Data<State<DB, KS>>,
+    Params(params): Params<(String,)>,
+) -> Result<KeyInfoJson, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let (addr_str,) = params;
+    let addr = Address::from_str(&addr_str)?;
+
+    let keystore = data.keystore.read().await;
+
+    let key_info = wallet::export_key_info(&addr, &*keystore)?;
+    Ok(KeyInfoJson(key_info))
+}
+
+/// Return whether or not a Key is in the Wallet
+pub(crate) async fn wallet_has<DB, KS>(
+    data: Data<State<DB, KS>>,
+    Params(params): Params<(String,)>,
+) -> Result<bool, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let (addr_str,) = params;
+    let addr = Address::from_str(&addr_str)?;
+
+    let keystore = data.keystore.read().await;
+
+    let key = wallet::find_key(&addr, &*keystore).is_ok();
+    Ok(key)
+}
+
+/// Import Keyinfo to the Wallet, return the Address that corresponds to it
+pub(crate) async fn wallet_import<DB, KS>(
+    data: Data<State<DB, KS>>,
+    Params(params): Params<(KeyInfoJson,)>,
+) -> Result<String, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let (KeyInfoJson(key_info),) = params;
+
+    let key = Key::try_from(key_info)?;
+
+    let addr = format!("wallet-{}", key.address.to_string());
+
+    let mut keystore = data.keystore.write().await;
+
+    keystore.put(addr, key.key_info)?;
+
+    Ok(key.address.to_string())
+}
+
+/// List all Addresses in the Wallet
+pub(crate) async fn wallet_list<DB, KS>(
+    data: Data<State<DB, KS>>,
+) -> Result<Vec<String>, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let keystore = data.keystore.read().await;
+    let addr_vec = wallet::list_addrs(&*keystore)?;
+    let ret = addr_vec.into_iter().map(|a| a.to_string()).collect();
+    Ok(ret)
+}
+
+/// Generate a new Address that is stored in the Wallet
+pub(crate) async fn wallet_new<DB, KS>(
+    data: Data<State<DB, KS>>,
+    Params(params): Params<(i8,)>,
+) -> Result<String, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+{
+    let (sig_type,) = params;
+    let mut keystore = data.keystore.write().await;
+    let key = match sig_type {
+        0 => wallet::generate_key(SignatureType::Secp256k1)?,
+        1 => wallet::generate_key(SignatureType::BLS)?,
```

Comment:
Sorry for the confusion before, this is actually 1,2 and would be best to use the actual type (as in the sig type being a param instead of an `i8`. `SignatureType` serializes as a u8 and can avoid matching here by putting that type into `generate_key`